file_transfer_agent.go

```diff
@@ -274,28 +323,80 @@ func (sfa *snowflakeFileTransferAgent) initFileMetadata() error {
 				stageLocationType: sfa.stageLocationType,
 				stageInfo:         sfa.stageInfo,
 			})
+		} else {
+			for i, fileName := range sfa.srcFiles {
+				fi, err := os.Stat(fileName)
+				if os.IsNotExist(err) {
+					return &SnowflakeError{
+						Number: ErrFileNotExists,
+						Message: fmt.Sprintf("file does not exist: %v",
+							fileName),
+					}
+				} else if fi.IsDir() {
+					return &SnowflakeError{
+						Number: ErrFileNotExists,
+						Message: fmt.Sprintf("not a file but a directory: %v",
+							fileName),
+					}
+				}
+				sfa.fileMetadata = append(sfa.fileMetadata, &fileMetadata{
+					name:              baseName(fileName),
+					srcFileName:       fileName,
+					srcFileSize:       int(fi.Size()),
+					stageLocationType: sfa.stageLocationType,
+					stageInfo:         sfa.stageInfo,
+				})
+				if len(sfa.encryptionMaterial) > 0 {
+					sfa.fileMetadata[i].encryptionMaterial = sfa.encryptionMaterial[0]
+				}
+			}
 		}
 
 		if len(sfa.encryptionMaterial) > 0 {
 			for _, meta := range sfa.fileMetadata {
 				meta.encryptionMaterial = sfa.encryptionMaterial[0]
 			}
 		}
+	} else if sfa.commandType == downloadCommand {
+		for _, fileName := range sfa.srcFiles {
+			if len(fileName) > 0 {
+				firstPathSep := strings.Index(fileName, "/")
```

Comment:
You are using the first path sep to get the file name?, and does '/' work for windows?

Reply:
We are parsing the result from the GS side which is in this format.